### PR TITLE
Maintenance/weekly upgrades

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- weekly requirement upgrades for ``packages``:
  - ``yarl`` PR#1836
  - ``pytest`` PR #1835
  - ``pytest-docker`` PR #1834 : removes constraints on ``attrs``. Both upgraded in all services as well!
  - ``docker-compose`` to 1.27.4
 
- Now pip-compile is done from a docker container instead of the developer host. Added draft of dev-container and used to pip-compile reqs: 
```cmd
cd scripts/requirements
make shell
# enters in container shell
$ cd scripts/requirements
$ make reqs
```

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
-# Docker
-**/Dockerfile
-**/docker-compose*
-
 # Byte-compiled / optimized / DLL files
 **/__pycache__
 *.py[cod]
+
+# virtualenv
+.venv
```
Comment (GitHK) [inline .dockerignore:11]: 👍

File: packages/service-library/requirements/_base.in [hunk 2]
```diff
@@ -16,6 +16,6 @@ werkzeug
 jsonschema
 prometheus_client
 tenacity
-attrs<20,>=19   # from pytest-docker==0.8.0
+attrs
```
Comment (GitHK) [inline packages/service-library/requirements/_base.in:19]: 👍

File: scripts/requirements/Dockerfile [hunk 3]
```diff
@@ -1,21 +1,29 @@
+# NOTE: This is a first step towards a devcontainer
+#       to perform operations like pip-compile or auto-formatting
+#       that preserves identical environment across developer machines
+#
```
Comment (GitHK) [inline scripts/requirements/Dockerfile:4]: really wanted this one!

Could it also be used for running tests in it?

File: scripts/requirements/Makefile [hunk 4]
```diff
@@ -48,6 +48,37 @@ reqs: _check_py36_version ## updates requirements of all package libraries
 	$(foreach p,${_input-requirements},touch $(p); $(MAKE_C) $(dir $(p)) reqs $(UPGRADE_OPTION);)
 
 
+# SEE https://medium.com/faun/set-current-host-user-for-docker-container-4e521cef9ffc
```
Comment (GitHK) [inline scripts/requirements/Makefile:51]: nice to know

**Review by GitHK — APPROVED** [review]
hope thing grows into a very useful toolkit

File: scripts/requirements/Dockerfile [hunk 5]
```diff
@@ -1,21 +1,29 @@
...
-# .dockerignore to include only target folders
-# mount osparc-simcore -> /work
-# cd script
-# make reqs
+RUN pip install \
+  black \
+  isort \
+  pip-tools \
+  bump2version \
+  rope
```
Comment (sanderegg) [inline scripts/requirements/Dockerfile:29]: :+1:

**Review by sanderegg — APPROVED** [review]
perfect!

**Pull request merged**